linera-views/src/key_value_store_view.rs

```diff
@@ -156,18 +156,28 @@ where
     C: Context + Send + Sync,
     ViewError: From<C::Error>,
 {
+    const NUM_INIT_KEYS: usize = 2 + ByteMapView::<C, u32>::NUM_INIT_KEYS;
+
     fn context(&self) -> &C {
         &self.context
     }
 
-    async fn load(context: C) -> Result<Self, ViewError> {
-        let key = context.base_tag(KeyTag::Hash as u8);
-        let hash = context.read_value(&key).await?;
-        let key = context.base_tag(KeyTag::TotalSize as u8);
-        let total_size = context.read_value(&key).await?.unwrap_or_default();
+    fn pre_load(context: &C) -> Result<Vec<Vec<u8>>, ViewError> {
+        let key_hash = context.base_tag(KeyTag::Hash as u8);
+        let key_total_size = context.base_tag(KeyTag::TotalSize as u8);
+        let mut v = vec![key_hash, key_total_size];
         let base_key = context.base_tag(KeyTag::Sizes as u8);
         let context_sizes = context.clone_with_base_key(base_key);
-        let sizes = ByteMapView::load(context_sizes).await?;
+        v.extend(ByteMapView::<C, u32>::pre_load(&context_sizes)?);
+        Ok(v)
+    }
+
+    fn post_load(context: C, values: &[Option<Vec<u8>>]) -> Result<Self, ViewError> {
+        let hash = from_bytes_opt(values.first().unwrap())?;
+        let total_size = from_bytes_opt(values.get(1).unwrap())?.unwrap_or_default();
```

Comment:
`from_bytes_opt` could be called `from_bytes_option` or `from_optional_bytes`.

More importantly, there should probably be a function for `from_bytes_opt( ... )?.unwrap_or_default()`. Maybe `from_optional_bytes_{with,or}_default`